lib/index.js

```diff
@@ -25,7 +25,26 @@ module.exports = class Rooftop {
           size: () => src.length
         }
       }
-      done()
+
+      return W.map(this.contentTypes, (ct) => {
+        if (ct.template) {
+          const name = ct.name
+          const data = compiler.options.spike.locals.rooftop[name]
+          const source = ct.template.path
+          const output = ct.template.output
+
+          return node.call(fs.readFile.bind(fs), path.join(source), 'utf8')
+            .then((jadeTpl) => {
+              return W.map(data, (item) => {
+                const renderedTpl = jade.render(jadeTpl, { item: item })
+                compilation.assets[output(item)] = {
+                  source: () => renderedTpl,
+                  size: () => renderedTpl.length
+                }
+              })
+            })
+          }
+      }).then(() => done())
```

Comment:
You need to finish the promise chain here so that errors are not suppressed. You can use `done` for this, with a callback to handle the error state. More info on [why this is the case here](https://github.com/cujojs/when/blob/master/docs/api.md#promisethen-vs-promisedone)